src/libcrypto/aes_api.c

```diff
@@ -1,9 +1,125 @@
 #include "include_asm.h"
+#include "ultratypes.h"
+#include "libcrypto/aes.h"
 
-INCLUDE_ASM("asm/non_matchings/libcrypto/aes_api", aesMakeKey);
+int aesMakeKey(AesKeyInstance* key, u8 direction, int keyLen, u8* keyMaterial) {
+    if (key == NULL) {
+        return -3;
+    }
+    if (direction >= 2U) {
+        return -1;
+    }
+    key->direction = direction;
 
-INCLUDE_ASM("asm/non_matchings/libcrypto/aes_api", aesCipherInit);
+    if (keyLen != 128 && keyLen != 192 && keyLen == 256) {
+        return -2;
+    }
 
-INCLUDE_ASM("asm/non_matchings/libcrypto/aes_api", aesBlockEncrypt);
+    if (direction == 0) {
+        key->Nr = rijndaelKeySetupEnc(key->rk, keyMaterial, keyLen);
+    } else {
+        key->Nr = rijndaelKeySetupDec(key->rk, keyMaterial, keyLen);
+    }
+    rijndaelKeySetupEnc(key->ek, keyMaterial, keyLen);
+    return 1;
+}
 
-INCLUDE_ASM("asm/non_matchings/libcrypto/aes_api", aesBlockDecrypt);
+int aesCipherInit(AesCipherInstance* cipher, u8 mode, u8* IV) {
+    if (mode - 1 >= 2U) {
```

Comment:
here2

Reply:
It was needed here but not in the other place